### PR TITLE
Use generated Grafana credentials, not default

### PR DESCRIPTION
**Pull request by andrewazores**
See #50 and #51 . This PR results in Grafana being configured to use a generated
password rather than the insecure default credentials for its normal HTTP Basic
auth scheme. The generated password is stored in a Secret, with the contents
written into the Grafana container's environment variables. These generated
credentials are also retrieved and used by the Operator itself when it performs
its startup configuration of the Grafana instance (creating the datasource
and dashboard).

### CODE REVIEW

File: pkg/controller/grafana/grafana_controller.go [hunk 1]
```diff
@@ -240,6 +252,12 @@ func (r *ReconcileGrafana) configureGrafanaDashboard(route *openshiftv1.Route) e
 	return nil
 }
 
+func GetCredentialedHostPathUrl(secret *corev1.Secret, host string, path string) string {
+	user := secret.Data["GF_SECURITY_ADMIN_USER"]
+	pass := secret.Data["GF_SECURITY_ADMIN_PASSWORD"]
+	return fmt.Sprintf("https://%s:%s@%s%s", strings.TrimSpace(string(user)), strings.TrimSpace(string(pass)), host, path)
```
Comment (ebaron) [inline pkg/controller/grafana/grafana_controller.go]: Quick question. Does the secret `Data` slice have Base64 encoded values in it? If so, would we need to decode them before inserting them into the URL?

Comment (andrewazores) [reply pkg/controller/grafana/grafana_controller.go]: Yea, I was a bit confused about that after reading the docs, but when I tested it manually I found that when reading/writing from the `Secrets` object, the data seemed to be automatically encoded/decoded. However when you view the contents of the secret with ex. `oc get -o json secret foo`, it's still encoded. I'll double check all that though.

Comment (andrewazores) [reply pkg/controller/grafana/grafana_controller.go]: Just checked by adding in a log statement when the grafana_controller reads the credentials here and they are indeed already decoded at this point, but when you do an `oc get secret` it's still encoded.

Comment (ebaron) [reply pkg/controller/grafana/grafana_controller.go]: Thanks for checking. The documentation was a bit unclear about that, and I couldn't find where the decoding happens. Seems good to go from me.

**Pull request merged**